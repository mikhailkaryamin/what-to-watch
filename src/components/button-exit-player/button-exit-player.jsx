import React from 'react';
import {connect} from 'react-redux';
import {func} from 'prop-types';

import {resetCurrentWatchedFilm} from '../../actions/actions.js';

const ButtonExitPlayer = (props) => {
  const {onButtonExitClick} = props;

  return (
    <button
      type="button"
      className="player__exit"
      onClick={() => {
        onButtonExitClick();
      }}
    >
      Exit
    </button>
  );
};

ButtonExitPlayer.propTypes = {
  onButtonExitClick: func.isRequired,
};

const mapDispatchToProps = (dispatch) => ({
  onButtonExitClick() {
    dispatch(resetCurrentWatchedFilm());
  }
});

export {
  ButtonExitPlayer,
};

export default connect(null, mapDispatchToProps)(ButtonExitPlayer);
