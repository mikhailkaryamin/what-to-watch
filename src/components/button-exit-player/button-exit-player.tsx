import React from 'react';
import {connect} from 'react-redux';
import {Link} from 'react-router-dom';

import {func} from 'prop-types';

import {AppRoute} from '../../const.js';
import {ActionCreator as ActionOptions} from '../../reducer/options/options.js';

const ButtonExitPlayer = (props) => {
  const {onButtonExitClick} = props;

  return (
    <Link
      to={AppRoute.ROOT}
    >
      <button
        type="button"
        className="player__exit"
        onClick={() => {
          onButtonExitClick();
        }}
      >
        Exit
      </button>
    </Link>

  );
};

ButtonExitPlayer.propTypes = {
  onButtonExitClick: func.isRequired,
};

const mapDispatchToProps = (dispatch) => ({
  onButtonExitClick() {
    dispatch(ActionOptions.resetFilm());
  }
});

export {
  ButtonExitPlayer,
};

export default connect(null, mapDispatchToProps)(ButtonExitPlayer);
