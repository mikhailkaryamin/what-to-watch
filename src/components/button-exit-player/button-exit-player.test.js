import React from 'react';
import {BrowserRouter as Router} from 'react-router-dom';

import renderer from 'react-test-renderer';

import {ButtonExitPlayer} from './button-exit-player.tsx';

describe(`button exit player`, () => {
  const wrapper = renderer
    .create(
        <Router>
          <ButtonExitPlayer
            onButtonExitClick={() => {}}
          />
        </Router>
    ).toJSON();

  test(`should render button exit player`, () => {
    expect(wrapper).toMatchSnapshot();
  });
});
