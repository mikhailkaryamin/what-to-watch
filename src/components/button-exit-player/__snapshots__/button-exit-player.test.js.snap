// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`button exit player should render button exit player 1`] = `
<a
  href="/"
  onClick={[Function]}
>
  <button
    className="player__exit"
    onClick={[Function]}
    type="button"
  >
    Exit
  </button>
</a>
`;
